tests/unit/s2n_tls13_secrets_test.c

```diff
@@ -488,5 +491,49 @@ int main(int argc, char **argv)
         };
     };
 
+    /* s2n_connection_export_secret */
+    {
+        /* Derives exporter secret on SERVER_FINISHED */
+        {
+            DEFER_CLEANUP(struct s2n_connection *conn = s2n_connection_new(S2N_SERVER),
+                    s2n_connection_ptr_free);
+            conn->secure->cipher_suite = &s2n_tls13_aes_128_gcm_sha256;
+            conn->actual_protocol_version = S2N_TLS13;
+            EXPECT_OK(s2n_connection_set_test_master_secret(conn, &test_secret));
+            EXPECT_BYTEARRAY_EQUAL(conn->secrets.version.tls13.exporter_master_secret,
+                    empty_secret, sizeof(empty_secret));
+
+            while (s2n_conn_get_current_message_type(conn) != SERVER_FINISHED) {
+                conn->handshake.message_number++;
+            }
+            EXPECT_OK(s2n_tls13_secrets_update(conn));
+
+            EXPECT_BYTEARRAY_NOT_EQUAL(conn->secrets.version.tls13.exporter_master_secret,
+                    empty_secret, sizeof(empty_secret));
+
+            // s2n_connection_tls_exporter requires us to finish the handshake.
+            // The above is needed since s2n_tls13_secrets_update will only
+            // initialize when it sees the SERVER_FINISHED frame.
+            while (s2n_conn_get_current_message_type(conn) != APPLICATION_DATA) {
+                conn->handshake.message_number++;
+            }
```

Comment:
We had to do this enough times that we added a testlib function: https://github.com/aws/s2n-tls/blob/main/tests/testlib/s2n_testlib.h#L78 Also, watch the comment formatting.